src/assets/scripts/client/aircraft/FlightManagementSystem/RouteModel.js

```diff
@@ -0,0 +1,1481 @@
+         */
+        this._previousLegCollection = [];
+
+        this.init(routeString);
+    }
+
+    /**
+     * Return the current `LegModel`
+     *
+     * @for RouteModel
+     * @property currentLeg
+     * @type {LegModel}
+     */
+    get currentLeg() {
+        if (this._legCollection.length < 1) {
+            throw new TypeError('Expected the route to contain at least one leg');
+        }
+
+        return this._legCollection[0];
+    }
+
+    /**
+     * Return the current `WaypointModel`
+     *
+     * @for RouteModel
+     * @property currentWaypoint
+     * @type {WaypointModel}
+     */
+    get currentWaypoint() {
+        return this.currentLeg.currentWaypoint;
+    }
+
+    /**
+     * Return the next `LegModel`, if it exists
+     *
+     * @for RouteModel
+     * @property nextLeg
+     * @type {LegModel}
+     */
+    get nextLeg() {
+        if (!this.hasNextLeg()) {
+            return null;
+        }
+
+        return this._legCollection[1];
+    }
+
+    /**
+     * Return the next `WaypointModel`, from current or future leg
+     *
+     * @for RouteModel
+     * @property nextWaypoint
+     * @type {WaypointModel}
+     */
+    get nextWaypoint() {
+        if (!this.hasNextWaypoint()) {
+            return null;
+        }
+
+        if (this.currentLeg.hasNextWaypoint()) {
+            return this.currentLeg.nextWaypoint;
+        }
+
+        return this.nextLeg.currentWaypoint;
+    }
+
+    /**
+     * Return an array of all waypoints in all legs of the route
+     *
+     * @for RouteModel
+     * @property waypoints
+     * @type {array<WaypointModel>}
+     */
+    get waypoints() {
+        return _reduce(this._legCollection, (waypointList, legModel) => {
+            return waypointList.concat(legModel.waypoints);
+        }, []);
+    }
+
+    // ------------------------------ LIFECYCLE ------------------------------
+
+    /**
+     * Initialize instance properties
+     *
+     * @for RouteModel
+     * @method init
+     * @param routeString {string}
+     * @chainable
+     */
+    init(routeString) {
+        this._legCollection = this._generateLegsFromRouteString(routeString);
+
+        return this;
+    }
+
+    /**
+     * Reset instance properties
+     *
+     * @for RouteModel
+     * @method reset
+     * @chainable
+     */
+    reset() {
+        this._legCollection = [];
+
+        return this;
+    }
+
+    // ------------------------------ PUBLIC ------------------------------
+
+    /**
+     * Merge the provided route model into this route model, if possible
+     *
+     * @for RouteModel
+     * @method absorbRouteModel
+     * @param routeModel {RouteModel}
+     * @return {array} [success of operation, response]
+     */
+    absorbRouteModel(routeModel) {
+        const firstWaypointName = _first(routeModel.waypoints).name;
+        const lastWaypointName = _last(routeModel.waypoints).name;
+        const routesConverge = this.hasWaypointName(lastWaypointName);
```

Comment:
these `const`s should be prepended by `is`:  `isConverging` and `isDiverging`.  Since these are both local to the method, we already know the context is that of the `route`, so I don't think that also needs to be included in the `const` name.